### PR TITLE
Remove unused (de)serializeAttributes() methods

### PR DESCRIPTION
**Pull request by rollerozxa**
Removes unused (de)serializeAttributes() methods located in the engine's GUI elements to coincide with IrrlichtMt PR https://github.com/minetest/irrlicht/pull/86.

### CODE REVIEW

**Review by sfan5 — APPROVED** [review]
Looks good.

File: src/gui/guiScene.cpp [hunk 1]
```diff
@@ -32,8 +32,6 @@ GUIScene::GUIScene(gui::IGUIEnvironment *env, scene::ISceneManager *smgr,
 
 	m_cam = m_smgr->addCameraSceneNode(0, v3f(0.f, 0.f, -100.f), v3f(0.f));
 	m_cam->setFOV(30.f * core::DEGTORAD);
-
-	m_smgr->getParameters()->setAttribute(scene::ALLOW_ZWRITE_ON_TRANSPARENT, true);
```
Comment (sfan5) [inline src/gui/guiScene.cpp:36]: hmm actually if these have to be hardcoded inside IrrlichtMt now that's worse IMO
guess `IAttributes` has to be kept then

Comment (rollerozxa) [reply src/gui/guiScene.cpp:36]: It would never be false anyways though?

Comment (sfan5) [reply src/gui/guiScene.cpp:36]: It removes the flexibility of being able to change these in the future if that's ever required.

Comment (rollerozxa) [reply src/gui/guiScene.cpp:36]: The setAllowZWriteOnTransparent() method itself in IVideoDriver (which is exposed to Minetest I think?) would probably be better to use if that need ever arises. And we don't use OBJ .mtl files but if we ever do, I'd assume it wouldn't be as easy as flipping a switch in Minetest anyways.

Comment (sfan5) [reply src/gui/guiScene.cpp:36]: `setAllowZWriteOnTransparent` exists but it cannot be used to influence the behavior of `CSceneManager::drawAll()` anymore:
grafik

**Review by sfan5 — APPROVED** [review]

**Review by rubenwardy — APPROVED** [review]

**Pull request merged**